## Summary:  
Software: gitlab  
## Fix:  
Commit: https://gitlab.com/gitlab-org/gitlab-foss/-/commit/d32802c76b37df936af5086e37bd93761c29799b  
Fix Time (of the Last Commit): 2018-07-16  
Author: Jamie Schembri  
Fix Pattern: Sanitization -> Redaction -> Reuse -> Keyword/RegexBased -> RedactionUtilDefined  
  
  
Fixing-commits Count: 1  
Total Change Size: +8 -3  
  
==========  
PR: https://gitlab.com/gitlab-org/gitlab-foss/-/merge_requests/20641  
Commit: https://gitlab.com/gitlab-org/gitlab-foss/-/commit/d32802c76b37df936af5086e37bd93761c29799b  
Change Size: +8 -3  
## Sample logs: Yes  
```  
RuntimeError: Error importing repository https://supersecret:credential@localhost/test-user-pass.git into root/test-user-pass - 13:CreateRepositoryFromURL: clone cmd wait: exit status 128  
```  
## Logging statements:  
### Source labels:  
Type of Sensitive Information:authentication data -> Sensitive Data  
### Sink labels:  
#### Sink 1:  
Logging call Function:  User-defined（error）  
Sensitiveness-Identifiable Source:  Same Modules & Cross Files（https://gitlab.com/gitlab-org/gitlab-foss/-/blob/d32802c76b37/app/models/project.rb#L614） -> Variable Names（credentials）  
Programming Language:Ruby  
```ruby  
# https://gitlab.com/gitlab-org/gitlab-foss/-/blob/d32802c76b37df936af5086e37bd93761c29799b/app/services/projects/import_service.rb#L17-27  
  
    def execute  
      add_repository_to_project  
  
      download_lfs_objects  
  
      import_data  
  
      success  
    rescue => e  
      error("Error importing repository #{project.import_url} into #{project.full_path} - #{e.message}") # HERE IS THE SINK 1  
    end  
  
  
```  
